src/lib/features/segment/segment-service.ts

```diff
@@ -151,10 +151,11 @@ export class SegmentService implements ISegmentService {
         await this.validateName(input.name);
         const segment = await this.segmentStore.create(input, auditUser);
 
+        const project = segment.project;
         await this.eventService.storeEvent(
             new SegmentCreatedEvent({
                 data: segment,
-                project: segment.project || 'no-project',
+                project: project,
```

Comment:
Not sure why you're extracting the variable here. It's only used once, and I don't think this adds any more clarity?

But if you want to do it, we can at least avoid doubling up:

```suggestion
                project,
```

Your call.